packages/react-core/src/components/Radio/Radio.tsx

```diff
@@ -110,41 +113,30 @@ class Radio extends React.Component<RadioProps, { ouiaStateId: string }> {
       />
     );
 
-    let labelRendered: React.ReactNode = null;
-    if (label && isLabelWrapped) {
-      labelRendered = <span className={css(styles.radioLabel, isDisabled && styles.modifiers.disabled)}>{label}</span>;
-    } else if (label) {
-      labelRendered = (
-        <label className={css(styles.radioLabel, isDisabled && styles.modifiers.disabled)} htmlFor={props.id}>
-          {label}
-        </label>
-      );
-    }
+    const wrapWithLabel = isLabelWrapped && !component;
```

Comment:
Just a thought, if the user passes `component="label"` then they'll end up with nested label elements. We could treat `component="label"` and `isLabelWrapped` the same.

```suggestion
    const wrapWithLabel = isLabelWrapped && (!component || component === 'label');
```

Reply:
This is a good idea! We _could_ possibly remove the `isLabelWrapped` prop as a breaking change in v6 and replace it entirely with just `component="label"`